src/auth.ts

```diff
@@ -0,0 +1,61 @@
+import { 
+	getAuth, 
+	createUserWithEmailAndPassword, 
+	signInWithEmailAndPassword,
+} from 'firebase/auth';
+import { Request, Response } from 'express';
+import app from './config/firebase.js';
+import { responses } from './constants/responses.js';
+import { errorCodes, statusCodes } from './constants/codes.js';
+
+export const signup = async(req: Request, res: Response): Promise<Response> => {
+	try {
+		const { email, password } = req.body;
+  
+		if (!email || !password) {
+			return res.status(statusCodes.unprocessableEntity).json({
+				email: responses.emailRequired,
+				password: responses.passwordRequired,
+			});
+		}
+  
+		const auth = getAuth(app);
+  
+		createUserWithEmailAndPassword(auth, email, password)
+			.then((user) => {
+				return res.status(statusCodes.created).json(user);
+			});
+	} catch (error) {
+		if (error.code === errorCodes.weakPassword) {
```

Comment:
res.status(error.code === errorCodes.weakPassword ? statusCodes.badRequest : .... )
   .json({ error: error.message });
   
or

create const for error.code === errorCodes.weakPassword ? statusCodes.badRequest :